tesseract/tesseract_planning/tesseract_process_managers/include/tesseract_process_managers/process_planning_server.h

```diff
@@ -0,0 +1,84 @@
+/**
+ * @file process_planning_server.h
+ * @brief A process planning server with a default set of process planners
+ *
+ * @author Levi Armstrong
+ * @date August 18, 2020
+ * @version TODO
+ * @bug No known bugs
+ *
+ * @copyright Copyright (c) 2020, Southwest Research Institute
+ *
+ * @par License
+ * Software License Agreement (Apache License)
+ * @par
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * @par
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef TESSERACT_PROCESS_MANAGERS_PROCESS_PLANNING_SERVER_H
+#define TESSERACT_PROCESS_MANAGERS_PROCESS_PLANNING_SERVER_H
+
+#include <tesseract_common/macros.h>
+TESSERACT_COMMON_IGNORE_WARNINGS_PUSH
+#include <memory>
+#include <string>
+#include <taskflow/taskflow.hpp>
+TESSERACT_COMMON_IGNORE_WARNINGS_POP
+
+#include <tesseract_motion_planners/core/profile_dictionary.h>
+
+#include <tesseract_process_managers/process_environment_cache.h>
+#include <tesseract_process_managers/taskflow_generator.h>
+#include <tesseract_process_managers/process_planning_request.h>
+#include <tesseract_process_managers/process_planning_future.h>
+
+namespace tesseract_planning
+{
+class ProcessPlanningServer
+{
+public:
+  using Ptr = std::shared_ptr<ProcessPlanningServer>;
+  using ConstPtr = std::shared_ptr<const ProcessPlanningServer>;
+
+  ProcessPlanningServer(EnvironmentCache::Ptr cache, size_t n = std::thread::hardware_concurrency());
```

Comment:
I went this direction because for the ROS side the environment is encapsulated in an environment monitor where I have a ROSEnvironmentCache which inherits from EnvironmentCache because I need to use the EnvironmentMonitors mutex for cloning the environment. Though I did not spend a lot of time looking into if there is a way to avoid the this.

Reply:
Yeah. I think you could still do this. I was just imaging the constructor that takes the environment to create the cache internally. For the ROS wrapper you would just use the constructor you already have.
```
ProcessPlanningServer(EnvironmentCache::Ptr cache, size_t n = std::thread::hardware_concurrency());
ProcessPlanningServer(Environment::ConstPtr environment, int cache_size = 1,  size_t n = std::thread::hardware_concurrency()) 
: ProcessPlanningServer(std::make_shared<EnvironmentCache>(environment, cache_size), n)
{
}
```

Reply:
But this is pretty minor so if you disagree, don't worry about it. We can add additional interfaces later. I assume this will have to be updated if/when we switch to using the CloneCache anyway

Reply:
Done